src/SFML/Window/Unix/KeyboardImpl.cpp

```diff
@@ -0,0 +1,799 @@
+    mapping.insert(std::make_pair("FK15", sf::Keyboard::ScanF15));
+    mapping.insert(std::make_pair("FK16", sf::Keyboard::ScanF16));
+    mapping.insert(std::make_pair("FK17", sf::Keyboard::ScanF17));
+    mapping.insert(std::make_pair("FK18", sf::Keyboard::ScanF18));
+    mapping.insert(std::make_pair("FK19", sf::Keyboard::ScanF19));
+    mapping.insert(std::make_pair("FK20", sf::Keyboard::ScanF20));
+    mapping.insert(std::make_pair("FK21", sf::Keyboard::ScanF21));
+    mapping.insert(std::make_pair("FK22", sf::Keyboard::ScanF22));
+    mapping.insert(std::make_pair("FK23", sf::Keyboard::ScanF23));
+    mapping.insert(std::make_pair("FK24", sf::Keyboard::ScanF24));
+    mapping.insert(std::make_pair("LMTA", sf::Keyboard::ScanLSystem));
+    mapping.insert(std::make_pair("RMTA", sf::Keyboard::ScanRSystem));
+    mapping.insert(std::make_pair("MUTE", sf::Keyboard::ScanVolumeMute));
+    mapping.insert(std::make_pair("VOL-", sf::Keyboard::ScanVolumeDown));
+    mapping.insert(std::make_pair("VOL+", sf::Keyboard::ScanVolumeUp));
+    mapping.insert(std::make_pair("STOP", sf::Keyboard::ScanStop));
+    mapping.insert(std::make_pair("REDO", sf::Keyboard::ScanRedo));
+    mapping.insert(std::make_pair("AGAI", sf::Keyboard::ScanRedo));
+    mapping.insert(std::make_pair("UNDO", sf::Keyboard::ScanUndo));
+    mapping.insert(std::make_pair("COPY", sf::Keyboard::ScanCopy));
+    mapping.insert(std::make_pair("PAST", sf::Keyboard::ScanPaste));
+    mapping.insert(std::make_pair("FIND", sf::Keyboard::ScanSearch));
+    mapping.insert(std::make_pair("CUT",  sf::Keyboard::ScanCut));
+    mapping.insert(std::make_pair("HELP", sf::Keyboard::ScanHelp));
+
+    mapping.insert(std::make_pair("I156", sf::Keyboard::ScanLaunchApplication1));
+    mapping.insert(std::make_pair("I157", sf::Keyboard::ScanLaunchApplication2));
+    mapping.insert(std::make_pair("I164", sf::Keyboard::ScanFavorites));
+    mapping.insert(std::make_pair("I166", sf::Keyboard::ScanBack));
+    mapping.insert(std::make_pair("I167", sf::Keyboard::ScanForward));
+    mapping.insert(std::make_pair("I171", sf::Keyboard::ScanMediaNextTrack));
+    mapping.insert(std::make_pair("I172", sf::Keyboard::ScanMediaPlayPause));
+    mapping.insert(std::make_pair("I173", sf::Keyboard::ScanMediaPreviousTrack));
+    mapping.insert(std::make_pair("I174", sf::Keyboard::ScanMediaStop));
+    mapping.insert(std::make_pair("I180", sf::Keyboard::ScanHomePage));
+    mapping.insert(std::make_pair("I181", sf::Keyboard::ScanRefresh));
+    mapping.insert(std::make_pair("I223", sf::Keyboard::ScanLaunchMail));
+    mapping.insert(std::make_pair("I234", sf::Keyboard::ScanLaunchMediaSelect));
+
+    return mapping;
+}
+
+////////////////////////////////////////////////////////////
+void ensureMapping()
+{
+    static bool isMappingInitialized = false;
+
+    if (isMappingInitialized)
+        return;
+
+    for (int i = 0; i < MaxKeyCode; ++i)
+        scancodeToKeycode[i] = NullKeyCode;
```

Comment:
Compiling with GCC 9.3 gave warnings here. The upper bound `MaxKeyCode` should be `sf::Keyboard::ScanCodeCount`. This line should be swapped with line 457.